File: 30-Days-of-Code/Day-6-Lets-Review/CSharp/solution.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        int num = int.Parse(Console.In.ReadLine());
        
        string firstHalf = "";
        string secondHalf = "";
        
        for (int i = 0; i < num; i++) {
            string s = Console.ReadLine();
            
            for (int j = 0; j < s.Length; j++) {
                if (j % 2 == 0)
                    firstHalf += s[j];
                else
                    secondHalf += s[j];
            }
            
            Console.WriteLine(firstHalf + " " + secondHalf);
            
            firstHalf = "";
            secondHalf = "";
        }
    }
}

File: 30-Days-of-Code/Day-6-Lets-Review/Perl/solution.pl
$n = int(<>);

for ($i = 0; $i < $n; $i++) {
    $s = <>;
    chomp($s);
    
    $firstHalf = "";
    $secondHalf = "";
    
    for ($j = 0; $j < length($s); $j++) {
        if ($j % 2 == 0) {
            $firstHalf = $firstHalf . substr($s,$j,1);
        } else {
            $secondHalf = $secondHalf . substr($s,$j,1);
        }
    }
    
    printf("%s %s\n", $firstHalf, $secondHalf);
}
